pwnlib/data/kctf/pow.py

```diff
@@ -0,0 +1,221 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file had been modified:
+# * Unnecessary code has been disabled
+# * It uses pwnlib logging instead of sys.stderr.write()
+# * The notice about installing gmpy2 has been moved into functions to make for a quieter import
+# * For Python 2 compatibility, random.randrange() is used if secrets.randbelow() is not available
+# * The 'can_bypass' mechanism has been removed to eliminate the dependence on ecdsa
+# * str(b, 'utf-8') has been replaced with six.ensure_str(b, 'utf-8')
+# * bytes(s, 'utf-8') has been replaced with six.ensure_binary(s, 'utf-8')
+# * n.to_bytes() has been replaced with packing.pack(n)
+# * int.from_bytes(b) has been replaced with packing.unpack(b, 'all')
+
+
+import base64
+# import os
+# import secrets
+# import socket
+# import sys
+# import hashlib
+import six
+from pwnlib.log import getLogger
+from pwnlib.util import packing
+
+log = getLogger(__name__)
+
+try:
+    import gmpy2
+    HAVE_GMP = True
+except ImportError:
+    HAVE_GMP = False
+    # sys.stderr.write("[NOTICE] Running 10x slower, gotta go fast? pip3 install gmpy2\n")
+
+try:
+    import secrets
+    HAVE_SECRETS = True
+except ImportError:
+    import random
+    HAVE_SECRETS = False
+
+VERSION = 's'
+MODULUS = 2**1279-1
+CHALSIZE = 2**128
+
+# SOLVER_URL = 'https://goo.gle/kctf-pow'
+
+def python_sloth_root(x, diff, p):
+    exponent = (p + 1) // 4
+    for i in range(diff):
+        x = pow(x, exponent, p) ^ 1
+    return x
+
+def python_sloth_square(y, diff, p):
+    for i in range(diff):
+        y = pow(y ^ 1, 2, p)
+    return y
+
+def gmpy_sloth_root(x, diff, p):
+    exponent = (p + 1) // 4
+    for i in range(diff):
+        x = gmpy2.powmod(x, exponent, p).bit_flip(0)
+    return int(x)
+
+def gmpy_sloth_square(y, diff, p):
+    y = gmpy2.mpz(y)
+    for i in range(diff):
+        y = gmpy2.powmod(y.bit_flip(0), 2, p)
+    return int(y)
+
+def sloth_root(x, diff, p):
+    if HAVE_GMP:
+        return gmpy_sloth_root(x, diff, p)
+    else:
+        log.warning_once("kctf-pow is running 10x slower, gotta go fast? pip3 install gmpy2")
+        return python_sloth_root(x, diff, p)
+
+def sloth_square(x, diff, p):
+    if HAVE_GMP:
+        return gmpy_sloth_square(x, diff, p)
+    else:
+        log.warning_once("kctf-pow is running 10x slower, gotta go fast? pip3 install gmpy2")
+        return python_sloth_square(x, diff, p)
+
+def encode_number(num):
+    size = (num.bit_length() // 24) * 3 + 3
+    # return str(base64.b64encode(num.to_bytes(size, 'big')), 'utf-8')
+    return six.ensure_str(base64.b64encode(packing.pack(num, size * 8, endian='big')), 'utf-8')
+
+def decode_number(enc):
+    # return int.from_bytes(base64.b64decode(bytes(enc, 'utf-8')), 'big')
+    return packing.unpack(base64.b64decode(six.ensure_binary(enc, 'utf-8')), 'all', endian='big')
```

Comment:
Do not use that `six` thing if an explicit way exists, please...
```suggestion
    return packing.unpack(base64.b64decode(enc.decode('utf-8')), 'all', endian='big')
```